tests/rptest/services/redpanda.py

```diff
@@ -1782,6 +1802,12 @@ def get_redpanda_pods_presorted(self):
 
         return active_rp_pods, inactive_rp_pods, unknown_rp_pods
 
+    def get_redpanda_statefulset(self):
```

Comment:
This only works on Azure or operator v2 clusters, right? Can you put an assert here that this is being called in a valid scenario?